### PR TITLE
ip2 for review

### PR DESCRIPTION
**Pull request by emarteca**

### CODE REVIEW

**Comment by emarteca** [comment]
@jkvaternik

File: Assignments/ip2/ip2.md [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+
+
+  /**
+   * Get the image contents for a specified poster session area (specified via poster session area controller)
+   * @param posterSessionArea the poster session area controller
+   * @returns a promise wrapping the contents of the poster session area's image (i.e. the string)
+   */
+  public async getPosterSessionAreaImageContents(
+    posterSessionArea: PosterSessionAreaController,
+  ): Promise<string>
```
Comment (jkvaternik) [inline Assignments/ip2/ip2.md:224]: Do they need to implement this or is it just here for docs/specifications? If they need to implement this, we need to update the handout file I believe.

File: Assignments/ip2/ip2.md [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+  public async incrementPosterSessionAreaStars(
+    posterSessionArea: PosterSessionAreaController,
+  )
+
+{% endhighlight %}
+</details>
+
+#### Grading for Task 1:
+Point break down for each of the implementation tasks:
+* Implement PosterSessionAreaController: 10 points
```
Comment (jkvaternik) [inline Assignments/ip2/ip2.md:240]: PosterSessionAreaController has some custom React hooks that might be good practice for students. Maybe we want to not give them that code for this class and have them implement it as well? Could help with balancing the points too.

**Review by jkvaternik — COMMENTED** [review]
Doc looks good! Seems pretty clear to me; I added some comments below about the IP that we could do to help balance the scale of points.

**Comment by iamabhutta** [comment]
-The PR preview only shows 3 paragraphs. 
Looks like mermaid is causing trouble. Can u embed the links from mermaid.live directly instead? I sent you what we used in Fa22 in discord

-Scores need to be out of 200 points, please update those from 150. Due to the way Canvas manages calculations it needs to at 200.

-Do we really has limit of 5 or 10 submissions on Gradescope. Please check the script and confirm if we provided the right numbers in IP2 instructions.

File: Assignments/ip2/ip2.md (hunk 2 above)
Comment (emarteca) [reply Assignments/ip2/ip2.md:240]: good idea -- added a section

File: Assignments/ip2/ip2.md [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+</details>
+
+Grading for Task 2:
+You do not need to write any tests for task 2. The handout contains all of the tests that our grading script will use.
+To run the tests, run `npm test PosterHooks` in the `frontend` directory.
+
+Point break down for each of the implementation tasks:
+* Implement PosterSessionAreaController.ts useStars: 10 points
+* Implement PosterSessionAreaController.ts useImageContents: 10 points
+* Implement PosterSessionAreaController.ts useTitle: 10 points
```
Comment (jkvaternik) [inline Assignments/ip2/ip2.md]: I think we just need these to add to 50. I double checked the handout and it looks good for this section too!

**Review by jkvaternik — COMMENTED** [review]
LGTM!

File: Assignments/ip2/ip2.md (hunk 3 above)
Comment (emarteca) [reply Assignments/ip2/ip2.md]: wow good catch

**Pull request merged**